crates/nu-cli/src/shell/filesystem_shell.rs

```diff
@@ -966,20 +966,21 @@ pub(crate) fn dir_entry_dict(
         }
     }
 
-    let name = if short_name {
-        filename.file_name().and_then(|s| s.to_str())
-    } else {
-        filename.to_str()
-    }
-    .ok_or_else(|| {
-        ShellError::labeled_error(
-            format!("Invalid file name: {:}", filename.to_string_lossy()),
-            "invalid file name",
-            tag,
-        )
-    })?;
-
-    dict.insert_untagged("name", UntaggedValue::string(name));
+    // let name = if short_name {
+    //     filename.file_name().and_then(|s| s.to_str())
+    // } else {
+    //     filename.to_str()
+    // }
+    // .ok_or_else(|| {
+    //     ShellError::labeled_error(
+    //         format!("Invalid file name: {:}", filename.to_string_lossy()),
+    //         "invalid file name",
+    //         tag,
+    //     )
+    // })?;
```

Comment:
I wonder if we should disable `name` and make a new column `short` when the user wants to use short names. They really aren't valid filenames anyways at that point, since you can't pass them into another command.

Reply:
I think I follow. You're saying that if `--short-names` is passed, just do it the 'old way' (the way prior to this change) and name that column `short` and remove the `name` column. Is that correct? At that point it would just be strings for file names and the `LS_COLORS` wouldn't work but maybe that's ok.  Do I understand your suggestion correctly?

Reply:
I guess another idea is to remove `--short-names` altogether and just not replace it. What that means is the name column will show the full path if the files aren't in the current directory like this:
![image](https://user-images.githubusercontent.com/343840/103162610-51641480-47b8-11eb-965d-c3b85244e43d.png)

Reply:
Yeah that sounds like what I was thinking

Reply:
ok. i think that would be pretty easy to do, i just hate adding this feature and it not work in some situations like with `-s`. i'd rather be consistent.  do you think it would be better to be consistent and just remove `-s` as an option altogether or would it be better to have colored file names work, except when you use `-s`. i mean, one can still do `ls | get name | path basename` and get just the filename only (which isn't colored BTW).